adapters/clients/client.go

```diff
@@ -73,7 +76,7 @@ func newRetryer() *retryer {
 }
 
 // n is the number of retries, work will always be called at least once.
-func (r *retryer) retry(ctx context.Context, n int, work func(context.Context) (bool, error)) error {
+func (r *retryer) retry(ctx context.Context, n int, work func(context.Context) (bool, error), retryLogEntry *logrus.Entry) error {
```

Comment:
Because this retry method gets a generic "work" function passed in as an argument, there isn't any useful information about which specific request is being retried. My proposed solution here is to pass in a *logrus.Entry which the caller can "attach" log fields to (eg HTTP method/URL), and then the retry method just adds its own retry info on top before logging. I'm curious if you think this is good enough, or if you know of a cleaner way to accomplish what I'm trying to do.

I think it would be overkill to rewrite the retry function and its callers so that the whole call stack has access to "more context", so that's why I tried this idea.

Reply:
I think having `method/URL` is good enough 👍 (assume we can identify collection and tenant from the URL?)

Reply:
They are HTTP requests, and IIRC we use the path to provide class name/shard name/ etc... so logging the http method+url should give us all the info we need.